### DISCUSSION TITLE
Github action step hanging

### DISCUSSION THREAD

**Discussion by ankushkatiyar** [community/community#25229] [Actions]
<p>Hi, i created a github action with single job only, that job contains multiple steps, but due to error in step it hanging untill timeout<br>
Can i catch that error to exit that step and proceed to next step or can i access exit code of last run command in run step and then exit that step ?</p>
<p>This is the error occured in step and then it start hanging</p>
<p>Error occured while creating tests { status: 500, value: { message: <code>Error occured while batching the tests {"message":"There're 2 active tunnels, please provide a tunnel identifier"}</code> } }</p>

**Comment by airtower-luna** [comment]
<p>The runner can’t detect if a running process is working as expected or hanging, you need to look at whatever runs in the step to fix it (and “crash instead of hang” would be a fix, then the runner would notice). If you’d like help with that, please post a link to an affected workflow run, or at least the relevant code.</p>
<p>To limit the impact meanwhile you can reduce <a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes"><code>timeout-minutes</code></a> for the step, so a hanging process is canceled earlier.</p>

**Comment by ankushkatiyar** [comment]
<p>Hi please check this url <a href="https://github.com/FoodSafetyAPP/react-cypress-lambdatest/runs/5751043885?check_suite_focus=true" class="inline-onebox" rel="noopener nofollow ugc">testing-5 · FoodSafetyAPP/react-cypress-lambdatest@fcdc7a0 · GitHub</a><br>
For now i reduced timeout to 5 minutes and then step terminated, you can see staus 500 when you open step “Execute lambdatest cli”<br>
I want to terminate step via programatically, not via timeout<br>
I check in documentation of github actions and by default bash shell using<br>
“* Fail-fast behavior using <code>set -eo pipefail</code> : Default for <code>bash</code> and built-in <code>shell</code> . It is also the default when you don’t provide an option on non-Windows platforms.”<br>
So i’m expecting if there any error in step then it will terminate automatically, but it will not doing like that<br>
You can check my workflow using this url<br>
<a href="https://github.com/FoodSafetyAPP/react-cypress-lambdatest/blob/fcdc7a0f5b81ba8ac386280c8e20002623f6237f/.github/workflows/cypress-lambdatest.yml" class="inline-onebox" rel="noopener nofollow ugc">react-cypress-lambdatest/cypress-lambdatest.yml at fcdc7a0f5b81ba8ac386280c8e20002623f6237f · FoodSafetyAPP/react-cypress-lambdatest · GitHub</a></p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="ankushkatiyar" data-post="3" data-topic="241797">
<div class="title">
<div class="quote-controls"></div>
 ankushkatiyar:</div>
<blockquote>
<p>So i’m expecting if there any error in step then it will terminate automatically, but it will not doing like that</p>
</blockquote>
</aside>
<p>It should terminate as failed as soon as one of the commands specified in the <code>run</code> step terminates with a non-zero exit code. The step you linked has exactly one <code>npm</code> command, so you’ll need to make sure that terminates. Error conditions <em>within</em> a command that don’t make it terminate won’t end the step.</p>

**Comment by ankushkatiyar** [comment]
<p>yeah i can understand this. But can i access exit code of run command and then terminate that step manually OR i can catch error using error handling methods if any in github actions ?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>A run step on Linux runs in Bash, you can use the regular Bash mechanisms to handle commands and their exit codes however you want. Just mind that by default that Bash runs with <code>set -e</code>, so it will terminate on the first non-zero exit code that isn’t handled. “Handling” means e.g. using the command as an <code>if</code> condition:</p>
<pre><code class="lang-bash">if your-command; then
    echo "command succeeded"
else
    echo "command failed"
fi
</code></pre>
